Dallia_Sintique/sistema_notas.py

```diff
@@ -0,0 +1,177 @@
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+
+        return codigo_opcao
+
+def obtem_nome(pergunta):
+  while True:
+    entrada_usuário = str(input(pergunta)).lstrip().rstrip()
+    if entrada_usuário.replace(" ", "").isalpha() and entrada_usuário != "":
+      break
+    else:
+      print("Digite um nome válido")
+  return entrada_usuário
+
+def obtem_nota(numero_nota, tipo):
+  status = 0
+  while status == 0:
+    try:
+        if tipo == 'n':
+            nota = float(input(f"Digite a nota {numero_nota} da aluna) (0 a 10): "))
+        else:
+            nota = float(input(f"Digite a nota de participação da aluna) (0 a 10): "))
+        if nota >= 0 and nota <= 10:
+            status = 1
+        else:
+            print("Digite uma nota válida entre 0 e 10")
+    except:
+      print(" Entrada inválida. Digite um número inteiro entre 0 e 10")
+  return nota
+
+def obtem_presenca(num_presenca):
+  contador = 0
+  while contador == 0:
+    presenca = input(f"Digite a presença {num_presenca} da aluna apenas com (True ou False): ").strip()
+    if presenca.lower() in ['true', 'false']:
+      presenca = presenca.lower() == 'true'
+      contador = 1
+    else:
+      print("Entrada inválida, informe a presença apenas com (True ou False)")
+  return presenca
+
+
+def incluir_nova_aluna():
+    print('Iniciando cadastro de nova aluna ')
+    notas = []
+    presencas = []
+
+    nome = obtem_nome(" Informe o nome da aluna para cadastro: ").title()
+    sobrenome = obtem_nome (" Informe o sobrenome da aluna para cadastro: ").title()
+    nome_sobrenome = (nome, sobrenome)
+
+    while True:
+        turma = (input ('Informe a turma da aluna com A / B ou C: ')).strip().upper()
+        if turma == "A" or turma == "B" or  turma == 'C':
+            break
+        else:
+            print('Entrada inválida. Informe uma turma com ( A / B ou C )')
+
+
+    for i in range (1, 4):
+        nota = obtem_nota(i, 'n')
+        notas.append(nota) 
+
+    for i in range (1, 6 ):
+        presenca = obtem_presenca(i)
+        presencas.append(presenca) 
+
+    participacao = obtem_nota(0,'p')
+
+    dataset[nome_sobrenome] = {
+        "Turma": turma ,
+        "Notas": notas,
+        "Presença": presencas ,
+        "Participação": participacao
+    }
+    print("Aluna cadastrada com sucesso ")
+
+
+def consultar_lista_alunas():
+    print(' Segue a lista de alunas cadastradas ')
+    if len(dataset) != 0:
+        for nome in dataset.keys(): 
+            print(nome[0])
```

Comment:
Apenas para curiosidade, podemos desempacotar a tupla dando nome aos respectivos campos, assim não precisariamos usar o índice da tupla :)

**Sugestão:**
```suggestion
        for nome, sobrenome in dataset.keys():
            print(f"{nome} {sobrenome}")
```